schema/deploy/function_search_all_facilities.sql

```diff
@@ -0,0 +1,64 @@
+-- Deploy ggircs-portal:function_search_all_facilities to pg
+-- requires: type_facility_application_status
+-- requires: view_ciip_application
+
+begin;
+
+create or replace function ggircs_portal.search_all_facilities(search_field text, search_value text, order_by_field text, direction text)
+
+    returns setof ggircs_portal.facility_application_status as
+        $function$
+            begin
+                if search_field is null or search_value is null
+                    then return query execute 'with applicationStatus as (
+                        select t.application_status as application_status, t.created_at, t.application_id as id
+                        from ggircs_portal.application_status t
+                        join (select application_id, max(created_at) as max_date
+                        from ggircs_portal.application_status
+                        group by application_id)a on a.application_id = t.application_id and a.max_date = t.created_at),
+
+                        applicationDetails as (
+                            select s.application_status as application_status, a.id as id, a.facility_id as facility_id
+                            from ggircs_portal.application as a join applicationStatus as s on a.id = s.id
+                        ),
+
+                        tempTable as (select f.*, ad.application_status as application_status, ad.id as application_id
+                        from ggircs_portal.facility as f
```

Comment:
This should only return the facilities that the user has access to. You can use the `session()` function to retrieve the `sub` field (which is a foreign key on `ciip_user.uuid`) and get the current user id. From there you need to do a join on the list of organisations the user has access to (via the `ciip_user_organisation` table), and only query the facilities from these organisations.

Reply:
to be done in #251